plugin/pkg/admission/webhook/admission.go

```diff
@@ -0,0 +1,169 @@
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/util/webhook"
+
+	"k8s.io/kubernetes/pkg/api"
+	admissionv1alpha1 "k8s.io/kubernetes/pkg/apis/admission/v1alpha1"
+	kubeapiserveradmission "k8s.io/kubernetes/pkg/kubeapiserver/admission"
+
+	// install the clientgo admissiony API for use with api registry
+	_ "k8s.io/kubernetes/pkg/apis/admission/install"
+)
+
+var (
+	groupVersions = []schema.GroupVersion{
+		admissionv1alpha1.SchemeGroupVersion,
+	}
+)
+
+func init() {
+	kubeapiserveradmission.Plugins.Register("GenericAdmissionWebhook", func(configFile io.Reader) (admission.Interface, error) {
+
+		var gwhConfig struct {
+			WebhookConfig GenericAdmissionWebhookConfig `json:"webhook"`
+		}
+
+		d := yaml.NewYAMLOrJSONDecoder(configFile, 4096)
+		err := d.Decode(&gwhConfig)
+
+		if err != nil {
+			return nil, err
+		}
+
+		plugin, err := NewGenericAdmissionWebhook(&gwhConfig.WebhookConfig)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return plugin, nil
+	})
+}
+
+// NewGenericAdmissionWebhook returns a generic admission webhook plugin.
+func NewGenericAdmissionWebhook(config *GenericAdmissionWebhookConfig) (admission.Interface, error) {
+	err := normalizeConfig(config)
+
+	if err != nil {
+		return nil, err
+	}
+
+	gw, err := webhook.NewGenericWebhook(api.Registry, api.Codecs, config.KubeConfigFile, groupVersions, config.RetryBackoff)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &GenericAdmissionWebhook{
+		Handler: admission.NewHandler(admission.Connect, admission.Create, admission.Delete, admission.Update),
+		webhook: gw,
+		rules:   config.Rules,
+	}, nil
+}
+
+// GenericAdmissionWebhook is an implementation of admission.Interface.
+type GenericAdmissionWebhook struct {
+	*admission.Handler
+	webhook *webhook.GenericWebhook
+	rules   []Rule
+}
+
+// Admit makes an admission decision based on the request attributes.
+func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) (err error) {
+	// The default rule will skip processing of the admission request
+	matched := DefaultRule
+	matchedIndex := -1
+
+	// Process all declared rules to attempt to find a match
+	for i, rule := range a.rules {
+		if Matches(rule, attr) {
+			glog.V(2).Infof("rule at index %d matched request", i)
+			matched = rule
+			matchedIndex = i
+
+			break
+		}
+	}
+
+	if matchedIndex == -1 {
+		glog.V(2).Infof("no rules matched request, using default")
+	}
+
+	// The matched rule skips processing this request
+	if matched.Type == Skip {
+		glog.V(2).Infof("rule explicitly allowed the request")
+		return nil
+	}
+
+	// Make the webhook request
+	request := admissionv1alpha1.NewAdmissionReview(attr)
+	response := a.webhook.RestClient.Post().Body(&request).Do()
+
+	// Handle webhook response
+	if err := response.Error(); err != nil {
+		return a.handleError(attr, matched.FailAction, err)
+	}
+
+	var statusCode int
+
+	if response.StatusCode(&statusCode); statusCode < 200 || statusCode >= 300 {
```

Comment:
I followed `ImagePolicyWebhook` so I don't have a preference.  I'm cool with using constants where possible, but it would look weird in my opinion if you had a constant on one side and a literal on the other.